### DISCUSSION TITLE
MEGA-POOL RFC

### DISCUSSION THREAD

**Discussion by cmc0619** [tagyoureit/nodejs-poolController#616] [Nixie]
Hi all,
I recently spoke with the Sequent Microsystems CEO and mentioned that his boards were being used in conjunction with _the_ best pool controller software around to make some pretty cool pool controllers and it'd be neat if we had a MEGA-POOL board just for us.  And he said "What would it look like?"

So I know my problems, personally, are having to choose between current based (4-20mA) and voltage based (0-10VDC) sensors and also the lack of 24V AC powering on the MEGA-BAS.  You have 48 pins to work with and RS-485 is there and not included in that 48 pin count.  What's on your MEGA-POOL wishlist?

Here was his request: 
"can you help with the ideal inputs and outputs, within the following restrictions: 

Power is 24VAC or DC, upper right. RS485 has 2 pins, upper left. 
Now we have 12 pins left on each side. We can install a mix and match of 2, 3, 4 or 6 pin connectors, assuming the total number of pins is 12. 
We can assign the pins to inputs or outputs, pretty much in any combination. "

**Comment by Crewski** [comment]
- I'd like to see dual RS-485 ports.  I have a Hayward SWG that is 9600 baud and a Hayward VS pump that is 19200
- at least 3 inputs for 10k thermistors (pool, air, solar)
- 3-4 relays with separate COMs
- ADC for pressure transducers (filter pressure)

Unfortunately everyone's setup are vastly different so it'll be tough to get a "one size fits all" board.  Thanks for working on this with them!

**Reply by cmc0619** [reply]
"Unfortunately everyone's setup are vastly different so it'll be tough to get a "one size fits all" board. Thanks for working on this with them!"

That's the problem I had of as well.  Then I realized I had a whole group of people in the ecosystem I could ask for their thoughts on the topic.   Going to try and get some ideas and hopefully the 95th percentile I'm thinking is the real 95th percentile.  So here I am polling the audience. 

Thanks for your response!

**Comment by rstrouse** [comment]
The MEGA-BAS is really a good start.
What I like
1. 24vac power
2. 8 universal inputs (selectable for 10k or dry contact)
3. RS485

However, I would like to see the outputs and RTC replaced with relays as many as one can get on the board.  It would be great if the connectors for those relays were 3-pin JST connectors for valve actuators using a common 24vac input.  If that were the case they only need to switch 1.5A and the connectors could be placed such that they did not consume so much edge space.   

The relay8 does a great job for circuit relays.  So a combination stack of a pool card and an 8 relay card would get you to the equivalent of an IntelliCenter i8(PS or D).  Add another relay 8 and you would be equivalent to an IntelliCenter i16(PS or D),

Heck if there was just a valve card that included 24vac switching for valve actuators with a common 24vac pole on the black wire as seen in the link below then it could be an input and valve board for a pool.  If one put 8 inputs and 12 JST-XH connectors on that board, you have more capacity for valve switching than is available for any pool controller on the market.
https://github.com/tagyoureit/nodejs-poolController/wiki/DIY-Standalone-Nixie-Pool-Controller#automated-valves

They have a 40A relay but unfortunately, I don't see it all that viable for switching split phase 240v equipment.  The mechanical link between the contactors is really the best way to go (3hp relay DPST) for things like 240v blowers, pumps, and filter relays are more akin to reliable switching.

**Reply by Telemachus1977** [reply]
Agreed.

But I think 8 inputs is an overkill.  Replace 4 of them with JST connector outputs for valves.  So one edge of the card will be inputs and valve control.

Then on the other side the card should be at least 4 relays for lights, etc with the screw terminals that the current relay boards have, but make the traces as large as possible to get more than 8 amps through the relays.

Also kill the modbus and 1 wire

**Comment by cmc0619** [comment]
"So a combination stack of a pool card and an 8 relay card would get you to the equivalent of an IntelliCenter i8(PS or D). Add another relay 8 and you would be equivalent to an IntelliCenter i16(PS or D),"

Have you met my friend the [Sequent Microsystems 16-relay board](https://sequentmicrosystems.com/collections/all-io-cards/products/sixteen-relays-8-layer-stackable-hat-for-raspberry-pi)?

I don't know if we'll be able to get JST connectors out of him.  I think he's trying to resuse the design used on the MEGA-BAS/IND boards. but I can ask!

Mihai also mentioned he'd like to stay away from selectable jumpers as they're apparently annoying to the general masses.  I feel like we're a pretty capable bunch of people though and this community would be able to handle it.

**Reply by Telemachus1977** [reply]
I have met your friend. The relays are only rated at 2 amps, and they are either normally open or normally closed. But not both.

**Reply by cmc0619** [reply]
Yeah, I guess that doesn't work so well for valve actuators...
